src/assets/Generator.Shared/ResponseWithError.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace Azure.Core
+{
+    internal static class ResponseWithError
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="ErrorResponse{T}"/> with the provided HTTP response and <see cref="RequestFailedException"/> exception.
+        /// </summary>
+        /// <param name="response">The HTTP response.</param>
+        /// <param name="exception"><see cref="RequestFailedException"/> exception.</param>
+        public static ErrorResponse<T> FromError<T>(Response response, RequestFailedException exception)
```

Comment:
@ellismg had suggested this approach in [this comment](https://github.com/Azure/autorest.csharp/issues/1334#issuecomment-924369290).  I believe the reason was to follow the existing response creation pattern, but if it only creates the error response instance, I agree it may not be needed.

Reply:
I was just following [`ValueResponse<T>`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Internal/ValueResponse.cs) and [`FromValue<T>`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Response.cs#L121-L124) pattern from Core but since we are using it only in generator I removed this class.

Reply:
Also, we follow this pattern in the generator as well. Check [ResponseWithHeaders](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Generator.Shared/ResponseWithHeaders.cs)

@annelo-msft @AlexanderSher

Reply:
yeah, these factory methods look excessive to me.